### PR TITLE
fix: re-enable the logging based on X-Request-Id header

### PR DESCRIPTION
**Pull request by FabioPinheiro**
Re-enable the logging based on X-Request-Id header

This was removed when we migrated to the latest version of ZIO-HTTP

### CODE REVIEW

**Review by mineme0110 — APPROVED** [review]

**Review by yshyn-iohk — APPROVED** [review]

File: logging.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Logging
+
+We want traceability from HTTP calls.
+- Each HTTP call needs to generate a call ID (preferable on the proxy). This ID must be should be pass to the scope of the ZIO application. So every log will mention this ID. This ID must also be returned to the user as a HTTP header.
```
Comment (petevielhaber) [inline logging.md:4]: - Each HTTP call needs to generate a call ID (preferably on the proxy). The ID must be within the scope of the ZIO application. So, every log will mention this ID. This ID must return to the user as an HTTP header.

File: logging.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- In case of errors and problem reports those IDs can be use by the user to get support.
```
Comment (petevielhaber) [inline logging.md:5]: - In case of errors and problem reports, the user can use those IDs to get support.

File: logging.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+Level 3 support can be provided to user during the logging retention policies. We user need to provide the value in the header `X-Request-Id` to get this support.
```
Comment (petevielhaber) [inline logging.md:7]: A user can receive Level 3 support during the logging retention policies. The user needs to provide the value in the header `X-Request-Id` to get this support.

File: logging.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+We have special annotations in the log so there is traceability between the different logs. 
+Here is the list of annotations and their meaning that we currently have:
+
+- `request-id` - Is the HTTP header `X-Request-Id` from the caller.
+  - If this header is missing, it is create by the APISIX https://apisix.apache.org/docs/apisix/plugins/request-id/. See the configuration how to enable in the file [apisixroute.yaml](https://github.com/input-output-hk/atala-prism-mediator/blob/eb6d822f125bea7b3da8f[…]0a378/infrastructure/charts/mediator/templates/apisixroute.yaml)
+- `msg_sha256` - Is the Hash (sha256) of the DID Comm message
+
+## Code
+
+To have a concise code we have created a Middleware that modifies the Annotations in the Scope before each execution of that endpoint, to include the trace ID of the request.
```
Comment (petevielhaber) [inline logging.md:20]: To have a concise code, we have created a middleware that modifies the annotations in the scope before each execution of that endpoint and includes the trace ID of the request.

File: logging.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+See code in file `TraceIdMiddleware` (`mediator/src/main/scala/io/iohk/atala/mediator/TraceIdMiddleware.scala`).
+
+## Logging Backend
+
+We use the Simple Logging Facade for Java (SLF4J) API to call the logging backend is determined at runtime.
```
Comment (petevielhaber) [inline logging.md:25]: We use the Simple Logging Facade for Java (SLF4J) API to call the logging backend determined at runtime.

**Review by petevielhaber — COMMENTED** [review]
Suggested edits

File: logging.md (hunk 1 above)
Comment (FabioPinheiro) [reply logging.md:4]: I'm creating a PR for this

File: logging.md (hunk 2 above)
Comment (FabioPinheiro) [reply logging.md:5]: I'm creating a PR for this

**Pull request merged**